go/common/headers.go

```diff
@@ -162,6 +162,24 @@ type CalldataRollupHeader struct {
 	ReOrgs [][]byte `rlp:"optional"` // sparse list of reorged headers - non null only for reorgs.
 }
 
+// PublicRollupMetadata contains internal rollup data that can be requested from the enclave.
+type PublicRollupMetadata struct {
```

Comment:
this is one way to marshal the metadata object.
The other way (which is what we've mostly done)  is to use the "converters" in the grpc classes).

You can have a look at `CrossChainMsg` , as an example